src/pages/ReportDetailsPage.tsx

```diff
@@ -100,12 +141,64 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
         return !pendingMember || pendingMember.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE ? accountID : [];
     });
 
-    const isGroupDMChat = useMemo(() => ReportUtils.isDM(report) && participants.length > 1, [report, participants.length]);
-
+    const caseID = useMemo((): CaseID => {
+        // 3. MoneyReportHeader
+        if (isMoneyRequestReport || isInvoiceReport) {
+            return CASES.MONEY_REPORT;
+        }
+        // 2. MoneyRequestHeader
+        if (isSingleTransactionView) {
+            return CASES.MONEY_REQUEST;
+        }
+        // 1. HeaderView
+        return CASES.DEFAULT;
+    }, [isInvoiceReport, isMoneyRequestReport, isSingleTransactionView]);
     const isPrivateNotesFetchTriggered = report?.isLoadingPrivateNotes !== undefined;
 
     const isSelfDM = useMemo(() => ReportUtils.isSelfDM(report), [report]);
 
+    const requestParentReportAction = useMemo(() => {
+        // 2. MoneyReport case
+        if (caseID === CASES.MONEY_REPORT) {
+            if (!reportActions || !transactionThreadReport?.parentReportActionID) {
+                return null;
+            }
+            return reportActions.find((action) => action.reportActionID === transactionThreadReport.parentReportActionID);
+        }
+        return parentReportAction;
+    }, [caseID, parentReportAction, reportActions, transactionThreadReport?.parentReportActionID]);
+
+    const isActionOwner =
+        typeof requestParentReportAction?.actorAccountID === 'number' && typeof session?.accountID === 'number' && requestParentReportAction.actorAccountID === session?.accountID;
+    const isDeletedParentAction = ReportActionsUtils.isDeletedAction(requestParentReportAction);
+
+    const moneyRequestReport = useMemo(() => {
+        if (caseID === CASES.MONEY_REQUEST) {
+            return parentReport;
+        }
+        return report;
+    }, [caseID, parentReport, report]);
+
+    const canModifyTask = Task.canModifyTask(report, session?.accountID ?? -1);
+    const shouldShowTaskDeleteButton =
+        isTaskReport &&
+        !isCanceledTaskReport &&
+        ReportUtils.canWriteInReport(report) &&
+        report.stateNum !== CONST.REPORT.STATE_NUM.APPROVED &&
+        !ReportUtils.isClosedReport(report) &&
+        canModifyTask;
+    const canDeleteRequest =
+        isActionOwner && (ReportUtils.canAddOrDeleteTransactions(moneyRequestReport) || ReportUtils.isTrackExpenseReport(transactionThreadReport)) && !isDeletedParentAction;
```

Comment:
Here, we are checking R`eportUtils.canAddOrDeleteTransactions(moneyRequestReport)` and `ReportUtils.isTrackExpenseReport(transactionThreadReport)` to determine if we should show the delete request button.
This fails for self-DM track because, for self-DM track, transactionThreadReport is undefined, which causes canAddOrDeleteTransactions to return false. This caused this Issue https://github.com/Expensify/App/issues/44334. 

We have handled the issue by adding a check for isTrackExpenseReport and isSelfDM for canDeleteRequest value. More details can be found in the selected proposal https://github.com/Expensify/App/issues/44334#issuecomment-2188094975